packages/storage/amplify_storage_s3_dart/lib/src/amplify_storage_s3_dart_impl.dart

```diff
@@ -113,6 +91,25 @@ class AmplifyStorageS3Dart extends StoragePluginInterface<
 
   AppPathProvider get _appPathProvider => dependencyManager.getOrCreate();
 
+  /// casts if [Object] is of type [T].
+  /// returns default if [Object] is null.
+  /// otherwise throws [S3Exception.invalidPluginOptions] exception.
+  static T tryCastOrDefaultIfNull<T>({
```

Comment:
Name is too generic and shouldn't be used outside validating plugin options. Would recommend `validatePluginOptions`

Reply:
had hard time to pick a name and looking for better options. however I'm not sure if `validatePluginOptions` reflects what the function does and used for.